framework/src/main/java/org/checkerframework/framework/type/TypeFromExpressionVisitor.java

```diff
@@ -83,8 +83,11 @@ public AnnotatedTypeMirror visitCompoundAssignment(
 
         // Recurse on the type of the variable.
         AnnotatedTypeMirror res = visit(node.getVariable(), f);
-        // TODO: why do we need to clear the type?
-        res.clearAnnotations();
+
+        if (f instanceof GenericAnnotatedTypeFactory
+                && ((GenericAnnotatedTypeFactory) f).getUseFlow()) {
+            res.clearAnnotations();
```

Comment:
Could you add a comment that explains why this is only necessary when flow-sensitive type refinement is enabled?

Reply:
The problem seems to be that getAnnotatedTypeLhs does not treat compound assignments specially.
It should do so, and then the call to cleanAnnotations can be removed entirely.